sites/hashdev/src/_pages/blog/10_error-stack-update-0-2.mdx

```diff
@@ -0,0 +1,185 @@
+
+```text
+experiment error: could not run experiment
+ ├╴examples/demo.rs:67:17
+ ├╴backtrace with 19 frames (3)
+ ├╴experiment 3 has no valid description
+ ╰╴unable to set up experiments
+
+ ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+ Backtrace No. 1
+    0: std::backtrace_rs::backtrace::libunwind::trace
+              at /rustc/8b705839cd656d202e920efa8769cbe43a5ee269/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
+    1: std::backtrace_rs::backtrace::trace_unsynchronized
+              at /rustc/8b705839cd656d202e920efa8769cbe43a5ee269/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
+    2: std::backtrace::Backtrace::create
+              at /rustc/8b705839cd656d202e920efa8769cbe43a5ee269/library/std/src/backtrace.rs:333:13
+    3: core::ops::function::FnOnce::call_once
+              at /rustc/8b705839cd656d202e920efa8769cbe43a5ee269/library/core/src/ops/function.rs:251:5
+    ...
+    18: _main
+```
+
+For a full guide to using hooks, please refer to [the `fmt` module documentation](https://docs.rs/error-stack/0.2.0/error_stack/fmt/index.html).
+
+## Related errors and multiple error sources
+
+`error-stack` v0.2 introduces support for related errors and multiple error sources, which can be invaluable when multiple errors occur at the same time, e.g. in loops or in multi-threaded applications. `Report`s are no longer a linear stream of events. You can now create a tree of different `Report`s, enabling the capture of multiple simultaneous errors without loosing information.
+
+This is also shown in the output:
+
+```text
+Error: experiment error: could not run experiment
+ ├╴examples/demo.rs:73:18
+ ├╴Unable to set up experiments
+ │
+ ├─▶ invalid experiment description
+ │   ├╴examples/demo.rs:43:10
+ │   ╰╴Experiment 2 could not be parsed
+ │
+ ╰┬▶ invalid digit found in string
+  │  ├╴examples/demo.rs:25:18
+  │  ╰╴"3o" could not be parsed as experiment
+  │
+  ╰▶ invalid digit found in string
+     ├╴examples/demo.rs:25:18
+     ╰╴"4a" could not be parsed as experiment
+
+ experiment error: could not run experiment
+ ├╴examples/demo.rs:67:17
+ ├╴Experiment 3 has no valid description
+ ╰╴Unable to set up experiments
+```
+
+Again, a huge thanks to Bilal Mahmoud ([@indietyp](https://github.com/indietyp)) for his work.
+
+## Improved documentation, `Termination`, `anyhow` and `eyre`
+
+We’ve made a few other improvements too:
+
+- The `error-stack` documentation has been greatly improved and we’re always open to more feedback ([create an Issue](https://github.com/hashintel/hash/tree/main/packages/libs/error-stack) or [join us in Discord](https://discord.com/invite/BPMrGAhjPh)).
+- `Termination` has been implemented for `Report`s: it will return the `ExitCode` provided by any `Context` if present, or `ExitCode::FAILURE` will be returned.
+- Compatibility for `anyhow` and `eyre` has been added to convert their types into `Report`.
```

Comment:
Should we mention this in the second paragraph?

Reply:
I think that this is a good idea unless @Alfred-Mountfield has an argument against it (it cannot be done from an external trait).

Reply:
I was convinced this was in a prior release (the initial impl was merged back in July https://github.com/hashintel/hash/pull/763). Turns out it wasn't in the initial release, we 100% should mention this in the intro I think.

Reply:
Added in 231d8025cb155270d39489b25080952af0d383bd

Reply:
Looks good. I still think we should consider 
> maybe emphasise that this should help people test out error-stack, as well as helping with gradual migration.

Reply:
Hmm, I think this would be placed in the section the comment was added at?

Reply:
103d7f7048b3dde851c40fa9fbb37bc706334964

Reply:
[Suggested a change](https://github.com/hashintel/hash/pull/1146/files#r985667589), thanks